uniref-rest/src/main/java/org/uniprot/api/uniref/service/UniRefEntryService.java

```diff
@@ -1,50 +1,34 @@
 package org.uniprot.api.uniref.service;
 
-import java.util.stream.Stream;
-
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
-import org.uniprot.api.common.repository.search.QueryResult;
-import org.uniprot.api.common.repository.search.SolrQueryRepository;
-import org.uniprot.api.rest.request.SearchRequest;
-import org.uniprot.api.rest.service.BasicSearchService;
-import org.uniprot.core.uniref.UniRefEntry;
-import org.uniprot.store.config.UniProtDataType;
-import org.uniprot.store.config.searchfield.common.SearchFieldConfig;
-import org.uniprot.store.config.searchfield.factory.SearchFieldConfigFactory;
-import org.uniprot.store.search.document.uniref.UniRefDocument;
+import org.uniprot.api.common.exception.ResourceNotFoundException;
+import org.uniprot.api.common.exception.ServiceException;
+import org.uniprot.api.uniref.repository.store.UniRefEntryStoreRepository;
+import org.uniprot.api.uniref.request.UniRefIdRequest;
 
 /**
  * @author lgonzales
  * @since 09/07/2020
  */
 @Service
-public class UniRefEntryService extends BasicSearchService<UniRefDocument, UniRefEntry> {
+public class UniRefEntryService {
 
-    private final SearchFieldConfig searchFieldConfig;
+    private final UniRefEntryStoreRepository entryStoreRepository;
 
     @Autowired
-    public UniRefEntryService(
-            SolrQueryRepository<UniRefDocument> repository, UniRefEntryConverter entryConverter) {
-        super(repository, entryConverter);
-        this.searchFieldConfig =
-                SearchFieldConfigFactory.getSearchFieldConfig(UniProtDataType.UNIREF);
-    }
-
-    @Override
-    protected String getIdField() {
-        return this.searchFieldConfig.getSearchFieldItemByName("id").getFieldName();
-    }
-
-    @Override
-    public QueryResult<UniRefEntry> search(SearchRequest request) {
-        throw new UnsupportedOperationException(
-                "UniRefEntryService does not support search, try to use UniRefLightSearchService");
+    public UniRefEntryService(UniRefEntryStoreRepository entryStoreRepository) {
+        this.entryStoreRepository = entryStoreRepository;
     }
 
-    @Override
-    public Stream<UniRefEntry> download(SearchRequest request) {
-        throw new UnsupportedOperationException(
-                "UniRefEntryService does not support download, try to use UniRefLightSearchService");
+    public UniRefEntryResult getEntity(String idValue, UniRefIdRequest idRequesst) {
+        try {
```

Comment:
Param name clusterId or unirefClusterId will improve the readability

Reply:
done